### DISCUSSION TITLE
git diff --stat "explorer"

### DISCUSSION THREAD

**Discussion by myitcv** [tpope/vim-fugitive#2215] [Q&A]
Hello again,

I have searched the documentation, either with the wrong terms, but so far not turned up anything. So would appreciate some guidance!

The scenario I am working with is as follows. For various reasons (one of which being a workaround for #2214) I want to browse the output of `git diff --stat $commit`. That is I want to explore the diff between the current working tree and `$commit`, drilling in by opening each file in the stat output and viewing the file diff with respect to `$commit`.

It feels like the way I want to navigate such a diff is by starting at the `git diff --stat` output. That is have a window that contains something like this:

```
 internal/cmd/preprocessor/cmd/parse.go                                         |  14 +-------
 internal/cmd/preprocessor/cmd/rootfile.go                                      | 133 +++++++++++++++++++++++-----------------------------------------------
 internal/cmd/preprocessor/cmd/scriptNode.go                                    |  27 ---------------
 internal/cmd/preprocessor/cmd/testdata/execute_err_upload_multiple_files.txtar |  49 --------------------------
 internal/cmd/preprocessor/cmd/testdata/execute_multistagescript.txtar          |  57 ------------------------------
 internal/cmd/preprocessor/cmd/uploadnode.go                                    |  37 --------------------
```

I would then like to be able to put the cursor on the filename in the diff, hit a keyboard shortcut and have that shortcut:

* open the file under the cursor
* run `:Gdiff $commit`

Am I holding things correctly here?

Is this in the spirit of what fugitive would look to support?

If it is, what's best way of achieving such a "browser" in Vim with fugitive?

Thanks

**Comment by tpope** [comment]
Have look at this output, from [the source to Git](/git/git) itself:

```
$ git diff --stat 53a06cf39b 1c78c78d25
 contrib/credential/netrc/.gitignore                           |  1 +
 contrib/credential/netrc/Makefile                             | 26 ++++++++++++++++++++++++--
 .../netrc/{git-credential-netrc => git-credential-netrc.perl} |  0
 3 files changed, 25 insertions(+), 2 deletions(-)
```

That ellipsis is concealing part of the file path. It is not possible to reliably get the file name. We could do it 99% of the time, but even settling for that, you're going to run into other challenges. How do you reliably determine what `$commit` to jump to? It's not impossible, but it's not something that I will be prioritizing.
